### DISCUSSION TITLE
Reading raw GNSS measurements from parsed data

### DISCUSSION THREAD

**Discussion by MaxiStocker** [semuconsulting/pyubx2#84] [Q&A]
Hello everyone, I am trying to read raw GNSS data (pseudoranges, phase count, doppler) from a parsed UBX message.  

<img width="1144" alt="image" src="https://user-images.githubusercontent.com/40360591/183581544-fc02b5f1-dfeb-48be-aa16-3921c8e7496b.png">
The pseudoranges, phase count, and doppler are in numbered variables, reading a specific one is not a problem, but how do I read all of them?

This is what I have so far:
```
from pyubx2 import UBXReader
from serial import Serial

stream = Serial('/dev/ttyACM0', 115200, timeout=3)
ubr = UBXReader(stream)
while True:
  try:
    for (rawData, parsedData) in ubr:
      if parsedData.identity == 'RXM-RAWX':
        # print(parsedData)
        print(parsedData.prMes_01)
  except NMEAParseError:
    print("parse error occurred.. continuing")
```

**Comment by semuadmin** [comment]
Hi @MaxiStocker 

pyubx2 decodes UBX data to a flat structure for convenience, with repeating attributes suffixed with an index value e.g. prMes_01, prMes_02, etc.

If you want to iterate through a repeating group as a list of attributes, you could use a construct along these lines:

```
prMesList = [] # list of prMes values from repeating group
size = msg.numMeas # size of repeating group
for i in range(size):
    prMes = getattr(msg, f"prMes_{i+1:02}")
    prMesList.append(prMes)
```
